internal/tfprotov6tov5/tfprotov6tov5_test.go

```diff
@@ -3265,6 +3288,131 @@ func TestValidateListResourceConfigResponse(t *testing.T) {
 	}
 }
 
+func TestListResourceRequest(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		in       *tfprotov6.ListResourceRequest
+		expected *tfprotov5.ListResourceRequest
+	}{
+		"nil": {
+			in:       nil,
+			expected: nil,
+		},
+		"all-valid-fields": {
+			in: &tfprotov6.ListResourceRequest{
+				Config:   &testTfprotov6DynamicValue,
+				TypeName: "test_list_resource",
+			},
+			expected: &tfprotov5.ListResourceRequest{
+				Config:   &testTfprotov5DynamicValue,
+				TypeName: "test_list_resource",
+			},
+		},
+	}
+
+	for name, testCase := range testCases {
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := tfprotov6tov5.ListResourceRequest(testCase.in)
+
+			if diff := cmp.Diff(got, testCase.expected); diff != "" {
+				t.Errorf("unexpected difference: %s", diff)
+			}
+		})
+	}
+}
+
+func TestListResourceServerStream(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
+		in       *tfprotov6.ListResourceServerStream
+		expected *tfprotov5.ListResourceServerStream
+	}{
+		"nil": {
+			in:       nil,
+			expected: nil,
+		},
+		"all-valid-fields": {
+			in:       testStreamv6,
+			expected: testStreamv5,
+		},
+	}
+
+	for name, testCase := range testCases {
+
+		t.Run(name, func(t *testing.T) {
+			t.Parallel()
+
+			got := tfprotov6tov5.ListResourceServerStream(testCase.in)
+
+			if got == nil {
+				if diff := cmp.Diff(got, testCase.expected); diff != "" {
+					t.Errorf("unexpected difference: %s", diff)
+				}
+			} else {
+				resultSlice := make([]tfprotov5.ListResourceResult, 0)
+				for res := range got.Results {
+					resultSlice = append(resultSlice, res)
+				}
+
+				expectedSlice := make([]tfprotov5.ListResourceResult, 0)
+				for res := range testCase.expected.Results {
+					expectedSlice = append(expectedSlice, res)
+				}
+
+				if len(expectedSlice) != len(resultSlice) {
+					t.Fatalf("expected iterator and result iterator lengths do not match")
+				}
+
+				for idx := range resultSlice {
+					if diff := cmp.Diff(resultSlice[idx], expectedSlice[idx]); diff != "" {
+						t.Errorf("unexpected difference: %s", diff)
+					}
+				}
+			}
+		})
+	}
+}
+
+func TestListResourceResult(t *testing.T) {
+	t.Parallel()
+
+	testCases := map[string]struct {
```

Comment:
💭 I'm thinking it's nice to add test cases for: identity is nil, resource is nil.

I convinced myself that the logic correctly handles these, but tests are forever 😃